src/contract.rs

```diff
@@ -684,22 +680,31 @@ fn cancel_ask(
     // is ask base a marker
     let is_base_restricted_marker = is_restricted_marker(&deps.querier, base.clone());
 
+    let mut response = Response::new();
+
     // return 'base' to owner, return converted_base to issuer if applicable
-    let mut response = Response::new()
-        .add_message(match is_base_restricted_marker {
-            true => {
-                transfer_marker_coins(size.into(), base, owner, env.contract.address.to_owned())?
-            }
-            false => BankMsg::Send {
+    match is_base_restricted_marker {
```

Comment:
looking at this, couldnt it be pulled out into a function as well?


```
pub fn add_transfer(
    mut response: Response,
    is_restricted: bool,
    amount: u128,
    denom: S,
    to: H,
    from: H,
    contract_address: H,
) -> Response {
    match is_restricted {
        true => {
            response = response.add_message(transfer_marker_coins(
                amount.into(),
                denom,
                to,
                from,
                contract_address,
            )?);
        }
        false => {
            response = response.add_message(BankMsg::Send {
                to_address: to.to_string(),
                amount: coins(u128::from(amount), denom),
            });
        }
    }
    response
}
```

called where ever needed, example:
```
response = add_transfer(
  &mut response,
  is_base_restricted_marker.to_owned(),
  size.into(),
  base.to_owned(),
  owner.to_owned(),
  env.contract.address.to_owned(),
  env.contract.address.to_owned(),
)
```

there are kind of a lot of usages of it in the `contract.rs` file

maybe out of scope for this PR though

Reply:
or 


```
pub fn get_transfer_msg(
    is_restricted: bool,
    amount: u128,
    denom: S,
    to: H,
    from: H,
    contract_address: H,
) -> impl Into<CosmosMsg<T>> {
    match is_restricted {
        true => {
            transfer_marker_coins(
                amount.into(),
                denom,
                to,
                from,
                contract_address,
            )?
        }
        false => {
            BankMsg::Send {
                to_address: to.to_string(),
                amount: coins(u128::from(amount), denom),
            }
        }
    }
}
```

and
```
response = response.add_message(
  get_transfer_msg(
    is_base_restricted_marker.to_owned(),
    size.into(),
    base.to_owned(),
    owner.to_owned(),
    env.contract.address.to_owned(),
    env.contract.address.to_owned(),
  )
)
```

Reply:
true. I'll follow this up in a separate PR

Reply:
could maybe make a separate shortcut task for it in the iteration?

Reply:
I just kept it in the same story
PR: https://github.com/FigureTechnologies/ats-smart-contract/pull/127/files